dev/core/src/com/google/gwt/dev/jjs/impl/GwtAstBuilder.java

```diff
@@ -537,13 +537,20 @@ public void endVisit(BreakStatement x, BlockScope scope) {
 
     @Override
     public void endVisit(CaseStatement x, BlockScope scope) {
+      assert x.isExpr == false : "Switch expressions not yet supported";
```

Comment:
Let's make this a Precondition instead, so it doesn't get skipped if asserts are disabled? 

(Note that this is feedback on my own original patch, I wrote this in the first draft, intending to do all testing with asserts enabled, but that isn't what a real user would be likely to do.)